streamer_config.go

```diff
@@ -68,6 +68,11 @@ type (
 		// You can do so using `new(StorageClientConfig)` in order to create a StorageClient
 		// with all possible configurations configured using their defaults as defined by this Go package.
 		StorageClient *StorageClientConfig
+
+		// BatchClient allows you to create a Batch API driven Streamer client.
+		// You can do so using `new(BatchClientConfig)` in order to create a BatchClient
+		// with all possible configurations configured using their defaults as defined by this Go package.
```

Comment:
Either this comment is wrong or your other comment is incorrect, as earlier you stated that you cannot create a default config as at least the source format is required to be known.

That being said, not sure I agree with that choice. I think we can make this into a default by for example defaulting to a Json-based datasource, and with the `Autodetect` being removed and instead implicitly inferred from the missing `bigquery.Schema`.